src/sqlite.rs

```diff
@@ -0,0 +1,513 @@
+             ON CONFLICT (wallet_name, txid, vout) DO UPDATE SET value = $4, script = $5",
+        )
+        .bind(wallet_name)
+        .bind(op.txid.to_string())
+        .bind(op.vout as i32)
+        .bind(txo.value.to_sat() as i64)
+        .bind(txo.script_pubkey.as_bytes())
+        .execute(&mut **db_tx)
+        .await?;
+    }
+
+    for (anchor, txid) in &changeset.anchors {
+        let block_hash = anchor.anchor_block().hash;
+        let anchor = serde_json::to_value(anchor)?;
+        sqlx::query(
+            "INSERT INTO anchor_tx (wallet_name, block_hash, anchor, txid) VALUES ($1, $2, jsonb($3), $4)
+             ON CONFLICT (wallet_name, block_hash, txid) DO UPDATE SET anchor = jsonb($3)",
+        )
+        .bind(wallet_name)
+        .bind(block_hash.to_string())
+        .bind(anchor)
+        .bind(txid.to_string())
+        .execute(&mut **db_tx)
+        .await?;
+    }
+
+    Ok(())
+}
+
+/// Select blocks.
+#[tracing::instrument]
+pub async fn local_chain_changeset_from_sqlite(
+    db_tx: &mut Transaction<'_, Sqlite>,
+    wallet_name: &str,
+) -> Result<local_chain::ChangeSet, BdkSqlxError> {
+    info!("local chain changeset from sqlite");
+    let mut changeset = local_chain::ChangeSet::default();
+
+    let rows = sqlx::query("SELECT hash, height FROM block WHERE wallet_name = $1")
+        .bind(wallet_name)
+        .fetch_all(&mut **db_tx)
+        .await?;
+
+    for row in rows {
+        let hash: String = row.get("hash");
+        let height: i32 = row.get("height");
+        let block_hash = BlockHash::from_str(&hash)?;
+        changeset.blocks.insert(height as u32, Some(block_hash));
+    }
+
+    Ok(changeset)
+}
+
+/// Insert blocks.
+#[tracing::instrument]
+pub async fn local_chain_changeset_persist_to_sqlite(
+    db_tx: &mut Transaction<'_, Sqlite>,
+    wallet_name: &str,
+    changeset: &local_chain::ChangeSet,
+) -> Result<(), BdkSqlxError> {
+    info!("local chain changeset to sqlite");
+    for (&height, &hash) in &changeset.blocks {
+        match hash {
+            Some(hash) => {
+                sqlx::query(
+                    "INSERT INTO block (wallet_name, hash, height) VALUES ($1, $2, $3)
+                     ON CONFLICT (wallet_name, hash) DO UPDATE SET height = $3",
+                )
+                .bind(wallet_name)
+                .bind(hash.to_string())
+                .bind(height as i32)
+                .execute(&mut **db_tx)
+                .await?;
+            }
+            None => {
+                sqlx::query("DELETE FROM block WHERE wallet_name = $1 AND height = $2")
+                    .bind(wallet_name)
+                    .bind(height as i32)
+                    .execute(&mut **db_tx)
+                    .await?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Collects information on all the wallets in the database and dumps it to stdout.
+#[tracing::instrument]
+pub async fn easy_backup(db: Pool<Sqlite>) -> Result<(), BdkSqlxError> {
```

Comment:
I didn't find an easy way to call this from outside the library, I guess because the module is private?

Reply:
I think we should remove these functions since they only print to stdout they belong in an example or new function on Store that returns list of wallet names. But can do that in a separate PR this one is already pretty big.